components/brave_shields/browser/ad_block_base_service.cc

```diff
@@ -195,6 +196,21 @@ bool AdBlockBaseService::TagExists(const std::string& tag) {
   return std::find(tags_.begin(), tags_.end(), tag) != tags_.end();
 }
 
+base::Optional<base::Value> AdBlockBaseService::HostnameCosmeticResources(
+        const std::string& hostname) {
+  return base::JSONReader::Read(
+          this->ad_block_client_->hostnameCosmeticResources(hostname));
```

Comment:
Callling this from UI thread was causing this quite nasty family of crashes https://github.com/brave/brave-browser/issues/10907 - just wanted to emphasize that we should super carefully review multithreaded code

Reply:
@iefremov yikes!  Are they sorted out now?  What would be helpful for @antonok-edm or I do from here to sort things out?

Reply:
@pes10k fixing it here https://github.com/brave/brave-core/pull/6250

Reply:
Gotcha, thanks a million @iefremov !